File: test-suite/handwritten-src/js/SetRecordTest.js
class SetRecordTest {
    constructor(module) {
        this.m = module;
    }
    testCppSetToJavaSet() {
        var jsSetRecord = this.m.testsuite.TestHelpers.getSetRecord();
        var jsSet = jsSetRecord.set;
        assertEq(3, jsSet.size);
        assertTrue(jsSet.has("StringA"));
        assertTrue(jsSet.has("StringB"));
        assertTrue(jsSet.has("StringC"));
    }

    testJavaSetToCppSet() {
        var jsSet = new Set();
        var iSet = new Set();
        jsSet.add("StringA");
        jsSet.add("StringB");
        jsSet.add("StringC");
        var jsSetRecord = {set: jsSet, iset: iSet};
        assertTrue(this.m.testsuite.TestHelpers.checkSetRecord(jsSetRecord));
    }
}
allTests.push(SetRecordTest);

File: test-suite/handwritten-src/swift/SetRecordTest.swift
import XCTest
@testable import TestSuite
@testable import TestSuiteBridge

final class SetRecordTest: XCTestCase {

    func testCppSetToSwiftSet() throws {
        let sSetRecord = try TestHelpers_statics.getSetRecord()
        let sSet = sSetRecord.set
        XCTAssertEqual(3, sSet.count)
        XCTAssertTrue(sSet.contains("StringA"))
        XCTAssertTrue(sSet.contains("StringB"))
        XCTAssertTrue(sSet.contains("StringC"))
    }

    func testSwiftSetToCppSet() throws {
        let sSetRecord = SetRecord(set: Set(["StringA", "StringB", "StringC"]), iset:Set([]))
        XCTAssertTrue(try TestHelpers_statics.checkSetRecord(sSetRecord))
    }
}
